code/modules/reagents/chemistry/reagents/drug_reagents.dm

```diff
@@ -464,3 +434,335 @@
 
 	M.adjustToxLoss(5 * REM * delta_time)
 	M.adjustOrganLoss(ORGAN_SLOT_LIVER, 3 * REM * delta_time)
+
+/datum/reagent/drug/mushroomhallucinogen
+	name = "Mushroom Hallucinogen"
+	description = "A strong hallucinogenic drug derived from certain species of mushroom."
+	color = "#E700E7" // rgb: 231, 0, 231
+	metabolization_rate = 0.2 * REAGENTS_METABOLISM
+	taste_description = "mushroom"
+	ph = 11
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
+	addiction_types = list(/datum/addiction/hallucinogens = 12)
+
+/datum/reagent/drug/mushroomhallucinogen/on_mob_life(mob/living/carbon/M, delta_time, times_fired)
+	if(!M.slurring)
+		M.slurring = 1 * REM * delta_time
+	SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "tripping", /datum/mood_event/high, name)
+	switch(current_cycle)
+		if(1 to 5)
+			if(DT_PROB(5, delta_time))
+				M.emote(pick("twitch","giggle"))
+		if(5 to 10)
+			M.Jitter(10 * REM * delta_time)
+			if(DT_PROB(10, delta_time))
+				M.emote(pick("twitch","giggle"))
+		if (10 to INFINITY)
+			M.Jitter(20 * REM * delta_time)
+			if(DT_PROB(16, delta_time))
+				M.emote(pick("twitch","giggle"))
+	..()
+
+/datum/reagent/drug/mushroomhallucinogen/on_mob_metabolize(mob/living/L)
+	. = ..()
+
+	if(!L.hud_used)
+		return
+
+	var/atom/movable/plane_master_controller/game_plane_master_controller = L.hud_used.plane_master_controllers[PLANE_MASTERS_GAME]
+
+	var/list/col_filter_identity = list(1,0,0,0, 0,1,0,0, 0,0,1,0, 0,0,0,1, 0.000,0,0,0)
+	var/list/col_filter_green = list(1,0,0,0, 0,1,0,0, 0,0,1,0, 0,0,0,1, 0.333,0,0,0)
+	var/list/col_filter_blue = list(1,0,0,0, 0,1,0,0, 0,0,1,0, 0,0,0,1, 0.666,0,0,0)
+	var/list/col_filter_red = list(1,0,0,0, 0,1,0,0, 0,0,1,0, 0,0,0,1, 1.000,0,0,0) //visually this is identical to the identity
+
+	game_plane_master_controller.add_filter("rainbow", 10, color_matrix_filter(col_filter_red, FILTER_COLOR_HSL))
+
+	for(var/filter in game_plane_master_controller.get_filters("rainbow"))
+		animate(filter, color = col_filter_identity, time = 0 SECONDS, loop = -1, flags = ANIMATION_PARALLEL)
+		animate(color = col_filter_green, time = 4 SECONDS)
+		animate(color = col_filter_blue, time = 4 SECONDS)
+		animate(color = col_filter_red, time = 4 SECONDS)
+
+	game_plane_master_controller.add_filter("psilocybin_wave", 1, list("type" = "wave", "size" = 2, "x" = 32, "y" = 32))
+
+	for(var/filter in game_plane_master_controller.get_filters("psilocybin_wave"))
+		animate(filter, time = 64 SECONDS, loop = -1, easing = LINEAR_EASING, offset = 32, flags = ANIMATION_PARALLEL)
+
+/datum/reagent/drug/mushroomhallucinogen/on_mob_end_metabolize(mob/living/M)
```

Comment:
please name `M` better

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names